### PR TITLE
add refresh to ibmqjob.result

### PR DESCRIPTION
**Pull request by luisg5**
### Summary

Addresses #466

### CODE REVIEW

File: qiskit/providers/ibmq/job/ibmqjob.py [hunk 1]
```diff
@@ -483,7 +486,7 @@ def _wait_for_completion(
 
         return self._status in required_status
 
-    def _retrieve_result(self) -> Result:
+    def _retrieve_result(self, refresh: bool = False) -> Result:
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/ibmqjob.py]: parameter not in the docstring. I'm surprise lint didn't catch this.

**Comment by jyu00** [comment]
Can you also update the changelog? Since you're changing a user facing method.

File: qiskit/providers/ibmq/job/ibmqjob.py (hunk 1 above)
Comment (luisg5) [reply qiskit/providers/ibmq/job/ibmqjob.py]: Found out docstrings are not required for private methods: https://www.python.org/dev/peps/pep-0008/#documentation-strings. As a result, pylint does not enforce them either. Must be why it wasn't caught 😅.

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,14 @@ The format is based on [Keep a Changelog].
 > - **Security**: in case of vulnerabilities.
 
 
+## [UNRELEASED]
+
+### Added
+
+- `IBMQJob.result()` now accepts an optional `refresh` parameter. If 
+  specified, the function re-queries the api for the results, rather
```
Comment (jyu00) [inline CHANGELOG.md]: nit - can you change `If specified` to `If refresh=true is specified`? Just to be clearer.

**Comment by jyu00** [comment]
One more thing. We should get into the habit of specifying the default value (e.g. `Default: foo`) in the docstring so users can get that information from the api doc (when it's available) instead of having to look at the code.

**Comment by luisg5** [comment]
> One more thing. We should get into the habit of specifying the default value (e.g. Default: foo) in the docstring so users can get that information from the api doc (when it's available) instead of having to look at the code.

Went back to add the default values for this PR. I'll make sure to do so in future ones as well 👍

**Pull request merged**